docs/command-line/promtool.md

```diff
@@ -321,6 +321,26 @@ Run labels query.
 
 
 
+##### `promtool query analyze`
+
+Run queries against your Prometheus to analyze the usage pattern of certain metrics.
+
+
+
+###### Flags
+
+| Flag | Description | Default |
+| --- | --- | --- |
+| <code class="text-nowrap">--server</code> | Prometheus server to query. | `http://localhost:9090` |
+| <code class="text-nowrap">--type</code> | Type of metric: classichistograms/nativehistograms. |  |
+| <code class="text-nowrap">--duration</code> | Time frame to analyze. | `1h` |
+| <code class="text-nowrap">--end</code> | Query end time (RFC3339 or Unix timestamp), defaults to now. |  |
+| <code class="text-nowrap">--step</code> | Query step size (duration). |  |
```

Comment:
I feel this should be renamed to scrape-interval , to indicate the actual use.

Reply:
IIRC it was called `scrape-interval` and I suggested to call it `step` to be consistent with existing `promtool` commands. 

But I threw in an idea above (bullet point 1 in the comment [above](https://github.com/prometheus/prometheus/pull/12331#pullrequestreview-1441887189)), which would render this parameter unnecessary. I'm not sure if @zenador picked it up or disproved it. I still have to look more closely at the code changes. (Sorry again for delays.)